include/jemalloc/internal/pa.h

```diff
@@ -85,7 +86,12 @@ struct pa_shard_s {
 	/* Allocates from a PAC. */
 	pac_t pac;
 
-	/* Allocates from a HPA. */
+	/*
+	 * We place a small extent cache in front of the HPA, since we intend
+	 * these configurations to use many fewer arenas, and therefore have a
+	 * higher risk of hot locks.
+	 */
+	sec_t hpa_sec;
```

Comment:
We should also flush the sec, when threads are terminating, e.g. similar to the last thread leaving an arena here: https://github.com/jemalloc/jemalloc/blob/5e41ff9b740258bddebcbd5575e1670a15f8b1ae/src/tcache.c#L721

Also is it true that, threads accessing the same shard all can flush all the sec shards, so we only need to worry about when the last thread is leaving?

Reply:
Done. I hooked this into the purge pathway generally, since that's probably what someone wants if they're purging manually, too.